test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -176,25 +176,18 @@
     (fill-drained-pool jrubies)
     result))
 
-(defn get-lock-from-pool
```

Comment:
Where the use of a `ReentrantReadWriteLock` would go away completely with this PR, I wanted to rewrite some of the tests to try to figure out whether the "write lock" is held or not by validating the behavior of public methods of the `JRubyPool` class rather than validating the state held internally for the "write lock" - e.g., the "write lock" must be "locked" if a non-writer thread borrower is unable to borrow instance while the "lock" is supposed to be held or must be "unlocked" if borrows from any threads are granted.